packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -23,47 +22,40 @@ const buildSafeArgumentsAccess = template(`
   let $0 = arguments.length > $1 ? arguments[$1] : undefined;
 `);
 
-function isSafeBinding(scope, node) {
-  if (!scope.hasOwnBinding(node.name)) return true;
-  const { kind } = scope.getOwnBinding(node.name);
-  return kind === "param" || kind === "local";
-}
-
 const iifeVisitor = {
-  ReferencedIdentifier(path, state) {
+  "ReferencedIdentifier|BindingIdentifier"(path, state) {
     const { scope, node } = path;
+    const { name } = node;
+
     if (
-      node.name === "eval" ||
-      !isSafeBinding(scope, node) ||
-      !isSafeBinding(state.scope, node)
+      name === "eval" ||
+      (scope.getBinding(name) === state.scope.parent.getBinding(name) &&
```

Comment:
Isn't this just a `!scope.hasOwnBinding` check?

Reply:
Nope, `state.scope` does not equal to `scope`.